content/manuals/build/building/cdi.md

```diff
@@ -0,0 +1,531 @@
+
+## Getting started
+
+To get started with CDI, you need to have a compatible environment set up. This
+includes having Docker v27+ installed with [CDI enabled](/reference/cli/dockerd.md#enable-cdi-devices)
+and Buildx v0.22+.
+
+You also need to create the [device specifications using JSON or YAML files](https://github.com/cncf-tags/container-device-interface/blob/main/SPEC.md#cdi-json-specification)
+in one of the following locations:
+
+* `/etc/cdi`
+* `/var/run/cdi`
+* `/etc/buildkit/cdi`
+
+> [!NOTE]
+> Location can be changed by setting the `specDirs` option in the `cdi` section
+> of the [`buildkitd.toml` configuration file](../buildkit/configure.md) if you
+> are using BuildKit directly. If you're building using the Docker Daemon with
+> the `docker` driver, see [Configure CDI devices](/reference/cli/dockerd.md#configure-cdi-devices)
+> documentation.
+
+## Building with a simple CDI specification
+
+Let's start with a simple CDI specification that injects an environment variable
+into the build environment and write it to `/etc/cdi/foo.yaml`:
+
+```yaml {title="/etc/cdi/foo.yaml"}
+cdiVersion: "0.6.0"
+kind: "vendor1.com/device"
+devices:
+- name: foo
+  containerEdits:
+    env:
+    - FOO=injected
+```
+
+Inspecting the `default` builder will show that the `vendor1.com/device` is
+detected as a device:
+
+```console
+$ docker buildx inspect
+Name:   default
+Driver: docker
+
+Nodes:
+Name:             default
+Endpoint:         default
+Status:           running
+BuildKit version: v0.23.2
+Platforms:        linux/amd64, linux/amd64/v2, linux/amd64/v3, linux/amd64/v4, linux/386
+Labels:
+ org.mobyproject.buildkit.worker.moby.host-gateway-ip: 172.17.0.1
+Devices:
+ Name:                  vendor1.com/device=foo
+ Automatically allowed: false
+GC Policy rule#0:
+ All:            false
+ Filters:        type==source.local,type==exec.cachemount,type==source.git.checkout
+ Keep Duration:  48h0m0s
+ Max Used Space: 658.9MiB
+GC Policy rule#1:
+ All:            false
+ Keep Duration:  1440h0m0s
+ Reserved Space: 4.657GiB
+ Max Used Space: 953.7MiB
+ Min Free Space: 2.794GiB
+GC Policy rule#2:
+ All:            false
+ Reserved Space: 4.657GiB
+ Max Used Space: 953.7MiB
+ Min Free Space: 2.794GiB
+GC Policy rule#3:
+ All:            true
+ Reserved Space: 4.657GiB
+ Max Used Space: 953.7MiB
+ Min Free Space: 2.794GiB
+```
+
+Now let's create a Dockerfile to use this device:
+
+```dockerfile
+# syntax=docker.io/docker/dockerfile-upstream:1.17-labs
+FROM busybox
+RUN --device=vendor1.com/device \
+  env | grep ^FOO=
+```
+
+Since [Dockerfile frontend v1.14.0-labs](../buildkit/dockerfile-release-notes.md#1140-labs),
+we can use the [`RUN --device` command](/reference/dockerfile.md#run---device)
+to request a device during a build. In this case we set `vendor1.com/device`
+device name which will request the first device available in the specification.
+In this case it will be `foo`.
```

Comment:
Perhaps something to correlate "first" with what was illustrated earlier in the manual;

```suggestion
device name which requests the first device available in the specification.
In this case it uses `foo`, which is the first device in our `/etc/cdi/foo.yaml`.
```